okta/resource_user.go

```diff
@@ -323,6 +324,18 @@ func resourceUserUpdate(d *schema.ResourceData, m interface{}) error {
 			}
 		}
 
+		if d.HasChange("group_memberships") {
+			err := updateGroupsOnUser(d.Id(), d.Get("group_memberships").([]interface{}), client)
+
+			if err != nil {
+				return err
+			}
+		}
+
+		if len(d.Get("group_memberships").([]interface{})) > 0 {
+			return nil
+		}
+
 		if err != nil {
```

Comment:
I might be missing context here but it looks like we may have redundant error checking. Ignore me if not. Also I am not sure why we return nil if there are no groups? Wouldn't we want to call read still?

Reply:
ah, good catch here.  This is leftover from when I stubbed out the `group_membership` stuff initially in the `Update` operation for `okta_user`.  The `return nil` conditional definitely needs to get removed and the `if err != nil` needs to get moved back to it's rightful/useful spot.